test/src/unittests/tonal/test_pitchsaliencefunctionpeaks.py

```diff
@@ -0,0 +1,174 @@
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks,mag_speaks)    
+        bins, values = PitchSalienceFunctionPeaks(minFrequency=100, maxFrequency=101)(calculatedPitchSalience)
+        self.assertEqualVector(bins,  [103.])
+        self.assertAlmostEqualVectorFixedPrecision(values, [0.13421774], 6)
+
+    # Similar to testRegression3Peaks, but considering minFrequency and maxFrequency
+    def testRegression3PeaksBelowMinFreq(self):
+        freq_speaks = [55, 100, 340] # length 1
+        mag_speaks = [1, 1, 1] # length 4
+        # For a single frequency 55 Hz with unitary  amplitude the 10 non zero salience function values are the following
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks,mag_speaks)    
+        bins, values = PitchSalienceFunctionPeaks(minFrequency=350, maxFrequency=351)(calculatedPitchSalience)
+        self.assertEqualVector(bins,  [320.])
+        self.assertAlmostEqualVectorFixedPrecision(values, [0.5], 6)
+
+    def testMinMaxFreqError(self): 
+        freq_speaks = [55, 100, 340] # length 1
+        mag_speaks = [1, 1, 1] # length 4
+        self.assertRaises(RuntimeError, lambda: PitchSalienceFunctionPeaks(minFrequency=150,maxFrequency=150)([]))
+
+    def testRegressionDifferentPeaks(self):
+        freq_speaks = [55, 85] 
+        mag_speaks = [1, 1] 
+        calculatedPitchSalience1 = PitchSalienceFunction()(freq_speaks,mag_speaks)
+        bins, values = PitchSalienceFunctionPeaks()(calculatedPitchSalience1)  
+        self.assertEqualVector(bins,  [0.,  75.] )
+        self.assertEqualVector(values, [1., 1.])    
+
+        bins, values = PitchSalienceFunctionPeaks()(calculatedPitchSalience1)                      
+        mag_speaks = [0.5, 2] 
+        calculatedPitchSalience2 = PitchSalienceFunction()(freq_speaks,mag_speaks)
+        bins, values = PitchSalienceFunctionPeaks()(calculatedPitchSalience2)  
+        self.assertEqualVector(bins,  [75., 0.] )
+        self.assertEqualVector(values, [2., 0.5])    
+
+    def testBelowReferenceFrequency1(self):
+        freq_speaks = [50] 
+        mag_speaks = [1] 
+        expectedPitchSalience = zeros(600)
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks,mag_speaks)        
+        bins, values = PitchSalienceFunctionPeaks()(calculatedPitchSalience)
+        self.assertEqualVector(bins, [])
+        self.assertEqualVector(values, [])    
+
+    # Same as above but using a non-default config parameter.
+    def testBelowReferenceFrequency2(self):
+        freq_speaks = [30] 
+        mag_speaks = [1] 
+        expectedPitchSalience = zeros(600)
+        calculatedPitchSalience = PitchSalienceFunction(referenceFrequency=40)(freq_speaks,mag_speaks)        
+        bins, values = PitchSalienceFunctionPeaks()(calculatedPitchSalience)
+        self.assertEqualVector(bins, [])
+        self.assertEqualVector(values, [])    
```

Comment:
Redundant? This has been covered in PitchSalienceFunction and PeakDetection should have a test for zero input.

Reply:
removed